pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -199,6 +203,48 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 				By(fmt.Sprintf("Get management cluster resources created by addons-manager for workload cluster %q on management cluster %q", clusterName, input.E2EConfig.ManagementClusterName))
 				clusterResources, err = getManagementClusterResources(ctx, mngclient, mngDynamicClient, mngAggregatedAPIResourcesClient, mngDiscoveryClient, namespace, clusterName, infrastructureName)
 				Expect(err).NotTo(HaveOccurred())
+
+				By(fmt.Sprintf("Upgrade workload cluster %q", clusterName))
+				currentTkrVersion, err := getCurrentTkrVersion(tkgCtlClient, clusterName, namespace)
+				Expect(err).ToNot(HaveOccurred())
+				nextAvailableTkrVersion, err := getNextAvailableTkrVersion(ctx, currentTkrVersion, mngClient)
+				Expect(err).ToNot(HaveOccurred())
+				err = tkgCtlClient.UpgradeCluster(tkgctl.UpgradeClusterOptions{
+					ClusterName: clusterName,
+					Namespace:   namespace,
+					TkrVersion:  nextAvailableTkrVersion,
+					SkipPrompt:  true,
+				})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() bool {
+					// Check should pass when the cluster's current tkr version is the same as next available version
+					// and the cluster is in running phase.
+					err, currentTkrVersion := getCurrentTkrVersion(tkgCtlClient, clusterName, namespace)
```

Comment:
woudl be nice if gertCurrentTkrVersion had the same signature as getNextAvailableTkrVersion